### ISSUE TITLE
SAM3 Image Inference Found 0 Objects

### ISSUE THREAD

**Issue by HaojunChen663** [facebookresearch/sam3#443]
Thanks for your excellent work. I have tried the example **image inference** script and changed the text prompt for several times:

```
import torch
#################################### For Image ####################################
from PIL import Image
from sam3.model_builder import build_sam3_image_model
from sam3.model.sam3_image_processor import Sam3Processor

# Load the model
bpe_path = "./sam3/sam3/assets/bpe_simple_vocab_16e6.txt.gz"
device = "cuda" if torch.cuda.is_available() else "cpu"
checkpoint_path = "/my_pretrained/sam3/sam3.pt"
model = build_sam3_image_model(bpe_path=bpe_path, device=device, checkpoint_path=checkpoint_path)
processor = Sam3Processor(model)

# Load an image
image_path = "./sam3/assets/images/test_image.jpg"
image = Image.open(image_path).convert("RGB")

inference_state = processor.set_image(image)
# Prompt the model with text
output = processor.set_text_prompt(state=inference_state, prompt="shoe")

# Get the masks, bounding boxes, and scores
masks, boxes, scores = output["masks"], output["boxes"], output["scores"]
print(f"Detected {masks.shape[0]} instances.")
print(f"Masks' shape: {masks.shape}")
```

But the model always returns **"Found 0 Objects"**, it seems that there is no error when loading the model and inferring. The output info is as followed:
```
model_builder.py:8: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
  import pkg_resources
Detected 0 instances.
Masks' shape: torch.Size([0, 1, 720, 1280])
Boxes' shape: torch.Size([0, 4])
```

Appreciate it if you could help me solve this issue!

**Comment by heyoeyo** [comment]
That seems strange that there are no results, and I don't see anything wrong with your code.
To make sure things are working you can set the model confidence to 0 to have it output everything. You can do this by modifying the processor line:
```python
processor = Sam3Processor(model, confidence_threshold=0.0)
```

That should result is there being 200 masks on the output. Then you could check which masks are highest scoring. When I run the example, I get 12 'shoe' detections. You can print out the top-12 scores with something like:

```python
best_scores_idx = torch.argsort(scores, descending=True)
print(scores[best_scores_idx[:12]])
```
For me the top-3 mask indexes are [135, 44, 184] for example, and the top-12 scores come out as:
`0.8074, 0.8060, 0.8053, 0.8038, 0.8022, 0.8019, 0.7984, 0.7853, 0.7838, 0.7823, 0.7792, 0.6525`
It seems like you must be getting different results, but maybe printing out this info could give clues for what's going on.
